lib/src/widgets/chat/audio/audio_seek_bar.widget.dart

```diff
@@ -0,0 +1,158 @@
+  AudioSeekBarWidgetState createState() => AudioSeekBarWidgetState();
+}
+
+class AudioSeekBarWidgetState extends State<AudioSeekBarWidget> {
+  late SliderThemeData sliderThemeData = SliderTheme.of(context).copyWith(
+    trackHeight: 2.0,
+  );
+
+  @override
+  Widget build(BuildContext context) {
+    return Stack(
+      children: [
+        bufferSlider(),
+        slider(),
+        audioDuration(),
+        audioPosition(),
+      ],
+    );
+  }
+
+  Widget bufferSlider() {
+    return Positioned.fill(
+      bottom: 12.0,
+      child: SliderTheme(
+        data: sliderThemeData.copyWith(
+          thumbShape: HiddenThumbComponentShape(),
+          activeTrackColor: Colors.red, //TODO: Buffer's color
+          inactiveTrackColor: widget.align == DSAlign.right
+              ? SystemColors.neutralDarkRooftop
+              : SystemColors.neutralLightBox,
+        ),
+        child: ExcludeSemantics(
+          child: Slider(
+            min: 0.0,
+            max: widget.duration.inMilliseconds.toDouble(),
+            value: min(widget.bufferedPosition.inMilliseconds.toDouble(),
+                widget.duration.inMilliseconds.toDouble()),
+            onChanged: (v) {},
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget slider() {
+    return Positioned.fill(
+      bottom: 12.0,
+      child: SliderTheme(
+        data: sliderThemeData.copyWith(
+          inactiveTrackColor: Colors.transparent,
+          thumbColor: widget.align == DSAlign.right
+              ? SystemColors.neutralLightSnow
+              : SystemColors.neutralDarkRooftop,
+          activeTrackColor: widget.align == DSAlign.right
+              ? SystemColors.primaryLight
+              : SystemColors.primaryNight,
+          thumbShape: const RoundSliderThumbShape(
+            enabledThumbRadius: 7.0,
+          ),
+        ),
+        child: Slider(
+          min: 0.0,
+          max: widget.duration.inMilliseconds.toDouble(),
+          value: min(widget.position.inMilliseconds.toDouble(),
+              widget.duration.inMilliseconds.toDouble()),
+          onChanged: (value) {
+            if (widget.onChanged != null) {
+              widget.onChanged!(Duration(milliseconds: value.round()));
+            }
+          },
+          onChangeEnd: (value) {
+            if (widget.onChangeEnd != null) {
+              widget.onChangeEnd!(Duration(milliseconds: value.round()));
+            }
+          },
+        ),
+      ),
+    );
+  }
+
+  Widget audioDuration() {
+    return Positioned(
+      right: 22.0,
+      bottom: 10.0,
+      child: Text(
+        RegExp(r'((^0*[1-9]\d*:)?\d{2}:\d{2})\.\d+$')
+                .firstMatch("${widget.duration}")
+                ?.group(1) ??
+            '${widget.duration}',
+        style: TextStyle(
+            color: widget.align == DSAlign.right
+                ? SystemColors.neutralLightSnow
+                : SystemColors.neutralDarkCity),
+      ),
+    );
+  }
+
+  Widget audioPosition() {
```

Comment:
This method could share the same "structure" with audioDuration(), since the only difference they have is that the audioDuration() uses widget.duration and audioPosition() uses widget.position.